### DISCUSSION TITLE
First draft wireframe for the Invenio upload form

### DISCUSSION THREAD

**Discussion by bjr70** [MESH-Research/knowledge-commons-repository#78] [Show and tell]
The [first set of wireframes](https://miro.com/app/board/uXjVMKi8gw0=/?share_link_id=622891158076) for the upload form can be found in Miro. The UX team will be discussing with Ian at 1pm.

**Comment by mikethicke** [comment]
I like the general approach here. There's a ton of information to collect but none of the screens feels too overwhelming. The progress bar is great. Most of the fields are pretty self-explanatory so I think most users are going to have a smooth experience.

A few thoughts / questions:

1. There is no abstract / description field. I think this is pretty valuable data for search and browsing the repository.
2. The contributor / creator distinction is not transparent to me (though I understand now what it is). I feel like there could be confusion here, especially as they are on different screens. How common is it to credit non-author contributions like this?
3. Some of the groupings of fields on screens seems arbitrary. (Language on the same screen as Subjects and Tags, or Funding on the same screen as DOI, etc.) This probably isn't a huge problem but see below.
4. There is a lot here, and it will probably grow. I wonder if it's possible to make some of these screens optional? If sections are named and there is a clearer logic to what fields are grouped in screens, maybe some of the less common options could just be skipped somehow? Like if the section tabs were down the left hand side and you just clicked to the ones you needed to fill out?
5. We probably need to think about recovering partially-completed deposits. What happens when you click "continue" on the first screen? Is the deposit created with that metadata?

**Reply by mikethicke** [reply]
Could Creators and Contributors be the same section, with a selector for that person's relation to the deposit (author, lead author, contributor, etc)?

**Reply by bjr70** [reply]
> Could Creators and Contributors be the same section, with a selector for that person's relation to the deposit (author, lead author, contributor, etc)?

They're separate fields in both Invenio and DataCite.

**Reply by bjr70** [reply]
So funny story, I forgot abstract/description. 😛 We've got a lot of revisions to make but this is hugely helpful, Mike. Sections will be named. Language with subjects made sense to me as a librarian as descriptors, but we will do some user testing on that. The sections will reorganize based on our discussion yesterday and I'm hoping to have a new set of revised wireframes done sometime next week. 

I would not save a partially completed record, TBH. Not sure how Invenio handles it though. I think that if someone bails halfway through and doesn't complete it it should not create a record.

**Reply by mikethicke** [reply]
On saving: 

I think it's up to how Ian implements it since Invenio's default deposit form is a single page. But suppose you're on like screen 4 and you accidentally refresh the page, or hit the back button, or your laptop dies. You might lose a lot of work.

On creator / contributor:

I didn't realize they were separate in Datacite, but makes sense I guess. We could still combine them in the UI if it makes the user experience better. It's possible other researchers are more familiar with this distinction than I am though.

**Reply by bjr70** [reply]
I was talking with Zoe and she did mention that it would be good if people could save and leave and come back and finish. I take back what I said above. Perhaps if it's not a metadata-only record and it sits out there for a period of time we could remind folks that it's still out there and occasionally cull abandoned records? Worth a thought. I also don't want to risk anyone losing work. That's infuriating. 

I'm open to combining creator/contributor in the UI if there's a good way to do it.  It would be less confusing for the user, I think.

**Reply by stephanievaskomsu** [reply]
Zoe and Mike's concerns are why I suggested a "Save and Continue" vs just "Next" schema...should have been more clear on that post-it

**Reply by monotasker** [reply]
I've created issue #85 to handle this question of saving partially complete deposits. It shouldn't be too difficult as long as the user completes the required fields in the first pass. If we want to allow saving *before* the required fields are done, then that's going to require digging into the Invenio internals.

**Reply by mikethicke** [reply]
You could make use of [local storage](https://www.robinwieruch.de/local-storage-react/) to store partially-completed form data. It could even update whenever a field changes or loses focus. Then if a user comes back to the deposit form they get a message like "You have a partially-completed deposit. Resume?" The Invenio deposit would never get created until the user finally clicks "Submit".

**Comment by mikethicke** [comment]
One more thing---on names---I wonder what best practice is these days for collecting name information, given that we now understand that different cultures have very different conventions around personal names.

**Reply by monotasker** [reply]
Good question. Bonnie's looking into what the most recent recommendations are. Datacite/Invenio have a single "name" field that holds the entire name, as well as fields for "given" and "family" names. I wonder whether in some situations we might want to let people choose to just use the single "name" field instead of collecting the two parts?

**Reply by bjr70** [reply]
It may be an option. Not sure though about how that might affect any of the data feeds that go out from the repository to Google Scholar, etc.

**Comment by stephanievaskomsu** [comment]
More dates thoughts from our convo today:

_Artwork: Juried Show_
Date of creation of artwork
Date of acceptance?
Dates of exhibition/show

_Music Performance_
Composition Date
First public performance
Additional performance(s)

**Comment by bjr70** [comment]
Closing. Discussion continues in https://github.com/MESH-Research/knowledge-commons-repository/discussions/91
